### PR TITLE
Add Audio clip filter

### PR DESCRIPTION
**Pull request by swordsreversed**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | fixes #issuenum |
| Related issues/PRs | #issuenum |
| License | MIT |
#### What's in this PR?

Adds an audio filter directly copied from the existing video clip filter.
#### Why?

Sometimes you need to clip audio and this is a useful addition.
#### Example Usage

``` php
$audio = $ffmpeg->open('audio.mp3');

// Now we can do
$audio->filters()->clip(FFMpeg\Coordinate\TimeCode::fromSeconds(0), FFMpeg\Coordinate\TimeCode::fromSeconds(60));



```

### CODE REVIEW

**Comment by Romain** [comment]
Hi @swordsreversed,
The code looks good, but shouldn't you write tests, too?
I think this would be a good complement to this pull request.

File: src/FFMpeg/Filters/Audio/AudioClipFilter.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (romainneutron) [inline src/FFMpeg/Filters/Audio/AudioClipFilter.php]: Missing php open tag

**Comment by swordsreversed** [comment]
Hi guys, will get onto these fixes asap.

**Comment by dubems** [comment]
Please, has this feature been integrated into the main codebase? I need to use this feature ASAP. Thanks

**Comment by Romain** [comment]
@dubems No. Please read @romainneutron's comment.

Your code lacks a php tag. Your code won't be merged until you apply these changes.

Moreover, to my opinion, you should write tests for this filter.

Finally, as a follow-up to my comment on issue #147, I'm wondering if the video clip is not already the solution...

File: src/FFMpeg/Filters/Audio/AudioClipFilter.php (hunk 1 above)
Comment (swordsreversed) [inline src/FFMpeg/Filters/Audio/AudioClipFilter.php]: fixed

**Comment by swordsreversed** [comment]
Hi sorry i've had to concentrate on other projects. I've fixed the tag, but currently don't have time to write tests. As for using the video clip filter for audio, seems somewhat counterintuitive and against good API practices to me...

**Comment by Romain** [comment]
Thanks for this change.
The tests are here to guarantee the stability of the bundle. Without them, this won't probably be merged.
I hear your point with regards to the readability of the filter and that it is counter-intuitive to use the video one for audio.

**Comment by Romain** [comment]
@swordsreversed, @taylorvc provided code for a filter which is the same as yours in issue #147.
He added 2 parameters to his request, which prevent the video to be re-encoded. I think that it may be interesting to edit your PR accordingly.

```php
    public function apply(Audio $audio, AudioInterface $format)
    {
        $commands = array('-ss', (string) $this->start);

        if ($this->duration !== null) {
          $commands[] = '-t';
          $commands[] = (string) $this->duration;
        }

        $commands[] = '-acodec';
        $commands[] = 'copy';

        return $commands;
    }
```

**Comment by swordsreversed** [comment]
@Romain thanks for that. I should have time next week to write tests and take a look at the code from @taylorvc.

**Comment by jens1o** [comment]
Are there any updates? Otherwise I could write the tests. But first, it needs to pull the latest master.

**Comment by swordsreversed** [comment]
@Romain I've added the copy flag from @taylorvc. @jens1o I've started writing the test but I'm not really a PHP guy and don't have the time to work out how to check the length of the clipped file in the test. Something like ffprobe doing:

```
ffprobe -i tests/files/Audio.mp3 -show_entries format=duration -v quiet -of csv="p=0" -sexagesimal
```

I'm actually using [node-fluent-ffmpeg](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg) now, sorry I couldn't get this merged. Good luck!

**Comment by Romain** [comment]
@jens1o, I'm not eager to merge a PR into master which hasn't tests written. This wouldn't be sustainable for the bundle, there are too much risks too introduce bugs.
@swordsreversed Could you please post here your the files that you created and/or modified so that @jens1o can integrate them into his fork and push a new PR with the tests?
Thanks

**Comment by swordsreversed** [comment]
@Romain all changes have been committed, you can grab em from the last commit. (btw if there was a test for the video clip filter this would not been a problem).

**Pull request closed without merging**